### PR TITLE
Initial commit of Base58/Base58Check support

### PR DESCRIPTION
**Pull request by hbs**

### CODE REVIEW

File: warp10/src/main/java/io/warp10/script/functions/B58TO.java [hunk 1]
```diff
@@ -0,0 +1,144 @@
...
+
+    if (!(top instanceof String)) {
+      throw new WarpScriptException(getName() + " operates on a STRING.");
+    }
+
+    byte[] decoded = decode((String) top);
+
+    if (check) {
+      if (decoded.length < prefix.length + 4) {
+        throw new WarpScriptException("Invalid length.");
```
Comment (ftence) [inline warp10/src/main/java/io/warp10/script/functions/B58TO.java:77]: Missing function name.
Also give more info like "base58 STRING is too short".

File: warp10/src/main/java/io/warp10/script/functions/B58TO.java [hunk 2]
```diff
@@ -0,0 +1,144 @@
...
+      }
+
+      for (int i = 0; i < prefix.length; i++) {
+        if (i > decoded.length || decoded[i] != prefix[i]) {
+          throw new WarpScriptException("Invalid prefix.");
```
Comment (ftence) [inline warp10/src/main/java/io/warp10/script/functions/B58TO.java:82]: Missing function name.

File: warp10/src/main/java/io/warp10/script/functions/B58TO.java [hunk 3]
```diff
@@ -0,0 +1,144 @@
...
+    } else if (this.check) {
+      prefix = (byte[]) top;
+      top = stack.pop();
+    }
+
+    if (!(top instanceof String)) {
+      throw new WarpScriptException(getName() + " operates on a STRING.");
+    }
+
+    byte[] decoded = decode((String) top);
```
Comment (ftence) [inline warp10/src/main/java/io/warp10/script/functions/B58TO.java:73]: Catch WSE to add function name.

File: warp10/src/main/java/io/warp10/script/functions/B58TO.java [hunk 4]
```diff
@@ -0,0 +1,144 @@
...
+      digest.update(hash, 0, hash.length);
+      digest.doFinal(hash, 0);
+
+      //
+      // Check trailer
+      //
+
+      for (int i = 0; i < 4; i++) {
+        if (hash[i] != decoded[decoded.length - 4 + i]) {
+          throw new WarpScriptException("Invalid checksum.");
```
Comment (ftence) [inline warp10/src/main/java/io/warp10/script/functions/B58TO.java:100]: Missing function name.

**Review by ftence — CHANGES_REQUESTED** [review]

**Comment by hbs** [comment]
PTAL

**Review by ftence — APPROVED** [review]

**Pull request merged**